sklearn/linear_model/_logistic.py

```diff
@@ -1042,10 +1042,9 @@ class LogisticRegression(LinearClassifierMixin,
     Parameters
     ----------
     penalty : {'l1', 'l2', 'elasticnet', 'none'}, default='l2'
-        Used to specify the norm used in the penalization. The 'newton-cg',
-        'sag' and 'lbfgs' solvers support only l2 penalties. 'elasticnet' is
-        only supported by the 'saga' solver. If 'none' (not supported by the
-        liblinear solver), no regularization is applied.
+        Used to specify the norm used in the penalization. If 'none',
+        no regularization is applied. See below for compatibility
+        with different solvers.
```

Comment:
We can be explicit by pointing to the solvers documentation:

```suggestion
        no regularization is applied. See the solvers documentation below
        for compatibility with different solvers.
```

Reply:
Thank you for the advice Thomas, this way is indeed much clearer. I've made the changes :)